src/javascript/detectors/os_command_injection/os_command_injection.js

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// {fact rule=os-command-injection@v1.0 defects=1}
+var express = require('express')
+var app = express()
+
+function osCommandInjectionNoncompliant() {
+    app.get('/user/:id', function (request, response) {
+        // Noncompliant: executing an operating system command from user-controlled data.
+        var stdout = execa.command("ls -la "+request.params.id)
+    })
+}
+// {/fact}
+
+
+// {fact rule=os-command-injection@v1.0 defects=0}
+var express = require('express')
+var app = express()
+
+function osCommandInjectionCompliant() {
+    app.get('/user/:id', (req, res) => {
+        // Compliant: command arguments are defined as elements of array to prevent injection.
+        var {stdout} =  execa("ls", ["-la", req.params.id])
```

Comment:
Some questions regarding `execa`:
- Why is this called without `.command()`, is there a difference?
- Why is `stdout` destructured but not in the noncompliant example?
- Based on the docs, it looks like this should return a `Promise` but I don't see any `await` or `.then()` expression. Can you confirm if this is valid?

Reply:
Yes there is a difference. `execa.command()` takes arguments as a string which is vulnerable and `execa()` takes arguments as elements of array which prevents injection.